<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Evan Teran</author>
 <class>DebuggerUI</class>
 <widget class="QMainWindow" name="DebuggerUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>edb</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QDisassemblyView" name="cpuView" native="true"/>
    </item>
    <item>
     <widget class="QListView" name="listView">
      <property name="font">
       <font>
        <family>Monospace</family>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_Help"/>
    <addaction name="separator"/>
    <addaction name="action_About"/>
    <addaction name="actionAbout_QT"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="action_Memory_Regions"/>
    <addaction name="action_Threads"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menu_Plugins">
    <property name="title">
     <string>&amp;Plugins</string>
    </property>
    <addaction name="action_Plugins"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menu_Options">
    <property name="title">
     <string>&amp;Options</string>
    </property>
    <addaction name="action_Configure_Debugger"/>
    <addaction name="actionApplication_Arguments"/>
    <addaction name="actionApplication_Working_Directory"/>
   </widget>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Open"/>
    <addaction name="action_Attach"/>
    <addaction name="action_Recent_Files"/>
    <addaction name="separator"/>
    <addaction name="actionE_xit"/>
   </widget>
   <widget class="QMenu" name="menu_Debug">
    <property name="title">
     <string>&amp;Debug</string>
    </property>
    <addaction name="action_Run"/>
    <addaction name="action_Pause"/>
    <addaction name="action_Restart"/>
    <addaction name="action_Detach"/>
    <addaction name="action_Kill"/>
    <addaction name="separator"/>
    <addaction name="action_Step_Into"/>
    <addaction name="action_Step_Over"/>
    <addaction name="separator"/>
    <addaction name="action_Run_Pass_Signal_To_Application"/>
    <addaction name="action_Step_Into_Pass_Signal_To_Application"/>
    <addaction name="action_Step_Over_Pass_Signal_To_Application"/>
    <addaction name="separator"/>
    <addaction name="actionRun_Until_Return"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_View"/>
   <addaction name="menu_Debug"/>
   <addaction name="menu_Plugins"/>
   <addaction name="menu_Options"/>
   <addaction name="menu_help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="registersDock">
   <property name="windowTitle">
    <string>Registers</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout">
     <item row="0" column="0">
      <widget class="QCategoryList" name="registerList">
       <property name="font">
        <font>
         <family>Monospace</family>
         <pointsize>9</pointsize>
        </font>
       </property>
       <column>
        <property name="text">
         <string>1</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dataDock">
   <property name="windowTitle">
    <string>Data Dump</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QGridLayout">
     <item row="0" column="0">
      <widget class="TabWidget" name="tabWidget">
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>00000000-00000000</string>
        </attribute>
        <layout class="QGridLayout"/>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="stackDock">
   <property name="windowTitle">
    <string>Stack</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout"/>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>ToolBar</string>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Pause"/>
   <addaction name="action_Step_Into"/>
   <addaction name="action_Step_Over"/>
   <addaction name="action_Run"/>
  </widget>
  <action name="action_Open">
   <property name="icon">
    <iconset resource="debugger.qrc">
     <normaloff>:/debugger/images/edb22-action-open.png</normaloff>:/debugger/images/edb22-action-open.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
  </action>
  <action name="action_Attach">
   <property name="text">
    <string>&amp;Attach</string>
   </property>
  </action>
  <action name="actionE_xit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="action_Memory_Regions">
   <property name="text">
    <string>&amp;Memory Regions</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="action_Single_Step">
   <property name="text">
    <string>&amp;Step Into</string>
   </property>
   <property name="shortcut">
    <string/>
   </property>
  </action>
  <action name="action_Run">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="debugger.qrc">
     <normaloff>:/debugger/images/edb22-action-run.png</normaloff>:/debugger/images/edb22-action-run.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Run</string>
   </property>
   <property name="shortcut">
    <string>F9</string>
   </property>
  </action>
  <action name="action_Pause">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="debugger.qrc">
     <normaloff>:/debugger/images/edb22-action-break.png</normaloff>:/debugger/images/edb22-action-break.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Pause</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="action_Restart">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Restart</string>
   </property>
  </action>
  <action name="action_Detach">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Detach</string>
   </property>
   <property name="iconText">
    <string>Detach</string>
   </property>
   <property name="toolTip">
    <string>Detach</string>
   </property>
  </action>
  <action name="action_Step_Into">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="debugger.qrc">
     <normaloff>:/debugger/images/edb22-action-stepi.png</normaloff>:/debugger/images/edb22-action-stepi.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Step Into</string>
   </property>
   <property name="shortcut">
    <string>F7</string>
   </property>
  </action>
  <action name="action_Step_Over">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="debugger.qrc">
     <normaloff>:/debugger/images/edb22-action-stepoveri.png</normaloff>:/debugger/images/edb22-action-stepoveri.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Step Over</string>
   </property>
   <property name="shortcut">
    <string>F8</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="action_Help">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Help</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="action_Toggle_Breakpoint">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Toggle Breakpoint</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="action_Configure_Debugger">
   <property name="text">
    <string>&amp;Configure Debugger</string>
   </property>
  </action>
  <action name="actionAbout_QT">
   <property name="text">
    <string>About &amp;QT</string>
   </property>
  </action>
  <action name="action_Breakpoint_Manager">
   <property name="text">
    <string>&amp;Breakpoint Manager</string>
   </property>
  </action>
  <action name="actionApplication_Arguments">
   <property name="text">
    <string>Application &amp;Arguments</string>
   </property>
  </action>
  <action name="actionRun_Until_Return">
   <property name="text">
    <string>Run &amp;Until Return</string>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
  </action>
  <action name="action_Step_Into_Pass_Signal_To_Application">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="debugger.qrc">
     <normaloff>:/debugger/images/edb22-action-stepi.png</normaloff>:/debugger/images/edb22-action-stepi.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Step Into (Pass Signal To Application)</string>
   </property>
   <property name="shortcut">
    <string>Shift+F7</string>
   </property>
  </action>
  <action name="action_Step_Over_Pass_Signal_To_Application">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="debugger.qrc">
     <normaloff>:/debugger/images/edb22-action-stepover.png</normaloff>:/debugger/images/edb22-action-stepover.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Step Over (Pass Signal To Application)</string>
   </property>
   <property name="shortcut">
    <string>Shift+F8</string>
   </property>
  </action>
  <action name="action_Run_Pass_Signal_To_Application">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="debugger.qrc">
     <normaloff>:/debugger/images/edb22-action-run.png</normaloff>:/debugger/images/edb22-action-run.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Run (Pass Signal To Application)</string>
   </property>
   <property name="shortcut">
    <string>Shift+F9</string>
   </property>
  </action>
  <action name="action_Recent_Files">
   <property name="text">
    <string>&amp;Recent Files</string>
   </property>
  </action>
  <action name="actionApplication_Working_Directory">
   <property name="text">
    <string>Application &amp;Working Directory</string>
   </property>
  </action>
  <action name="action_Kill">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Kill</string>
   </property>
  </action>
  <action name="action_Plugins">
   <property name="text">
    <string>&amp;Plugins</string>
   </property>
  </action>
  <action name="action_Threads">
   <property name="text">
    <string>&amp;Threads</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCategoryList</class>
   <extends>QTreeWidget</extends>
   <header>QCategoryList.h</header>
  </customwidget>
  <customwidget>
   <class>QDisassemblyView</class>
   <extends>QWidget</extends>
   <header>QDisassemblyView.h</header>
  </customwidget>
  <customwidget>
   <class>TabWidget</class>
   <extends>QTabWidget</extends>
   <header>TabWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="debugger.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionE_xit</sender>
   <signal>triggered()</signal>
   <receiver>DebuggerUI</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>494</x>
     <y>437</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
